examples/core/core_input_gamepad.nim

```diff
@@ -0,0 +1,179 @@
+#[
+*
+*   raylib [core] example - Gamepad input
+*
+*   NOTE: This example requires a Gamepad connected to the system
+*         raylib is configured to work with the following gamepads:
+*                - Xbox 360 Controller (Xbox 360, Xbox One)
+*                - PLAYSTATION(R)3 Controller
+*         Check raylib.h for buttons configuration
+*
+*   This example has been created using raylib 2.5 (www.raylib.com)
+*   raylib is licensed under an unmodified zlib/libpng license (View raylib.h for details)
+*
+*   Copyright (c) 2013-2019 Ramon Santamaria (@raysan5)
+*   Adapted in 2021 by jseb
+*
+********************************************************************************************]#
+
+import nimraylib_now
+
+# NOTE: Gamepad name ID depends on drivers and OS
+when defined(PLATFORM_RPI):
+  const XBOX360_NAME_ID = "Microsoft X-Box 360 pad"
+else:
+  const XBOX360_NAME_ID = "Xbox 360 Controller"
+
+const
+  PS3_NAME_ID = "PLAYSTATION(R)3 Controller"
+  XBOX360_LEGACY_NAME_ID = "Xbox Controller"
+  screenWidth = 800
+  screenHeight = 450
+
+type tex_names = enum PS3Pad, XboxPad
+var texs:array[tex_names,Texture2D]
+
+
+proc init() =
+  setConfigFlags(MSAA_4X_HINT)  # Set MSAA 4X hint before windows creation
+
+  initWindow(screenWidth, screenHeight, "raylib [core] example - gamepad input")
+
+  texs[Ps3Pad] = loadTexture("resources/ps3.png")
+  texs[XboxPad] = loadTexture("resources/xbox.png")
+
+  setTargetFPS( 60)
+
+proc pad_draw_xbox
+proc pad_draw_psx
+
+
+proc main() =
+  while not windowShouldClose():
+      beginDrawing:
+        clearBackground( RAYWHITE)
+        if isGamepadAvailable( 0):
+          let gamepad_name = getGamePadName( 0)
+          drawText( "GP1: " & $gamepad_name, 10, 10, 10, BLACK);
+
+          case $gamepad_name:
+            of XBOX360_NAME_ID, XBOX360_LEGACY_NAME_ID:
+              pad_draw_xbox()
+            of PS3_NAME_ID:
+              pad_draw_psx()
+            else:
+              # well … for me it will be a PS pad view as a « USB Gamepad »
+              pad_draw_psx()
+
+          drawText( "DETECTED AXIS: " & $getGamepadAxisCount(0), 10, 50, 10, MAROON)
+          for i in 0..getGamepadAxisCount(0):
+            #drawText(TextFormat("AXIS %i: %.02f", i, GetGamepadAxisMovement(0, i)), 20, 70 + 20*i, 10, DARKGRAY);
+            drawText( "AXIS " & $i & " : " & $getGamepadAxisMovement(0, i), 20, 70 + 20*i, 10, DARKGRAY)
```

Comment:
Can we delete this commented line or it is here for information?

Reply:
Original line in C
```c
DrawText(TextFormat("AXIS %i: %.02f", i, GetGamepadAxisMovement(0, i)), 20, 70 + 20*i, 10, DARKGRAY);
```

uses formatting `"AXIS %i: %.02f"` similar to C's `printf` or Nim's `strformat` module.

`"AXIS " & $i & " : " & $getGamepadAxisMovement(0, i)` this looks like concatenation of strings, not the formatting via `%i` and other variables. We might want to keep it as close as possible to the original examples

Reply:
Yes i had forgotten to remove the C pattern, should be deleted.

And for concatenation, it was for avoiding an include to strformat, but thinking twice i realize that TextFormat is a part of the raylib API. May be we may meet halfway, and using strformat which will be cleaner than forcing conversions to C types with textFormat ?

Reply:
I think both options have their merits, either way will be great